modmesh/onedim/euler1d.py

```diff
@@ -302,16 +302,27 @@ def calc_internal_energy(self, pressure, density):
     def calc_entropy(self, pressure, density):
         return pressure / (density ** self.gamma)
 
-    def build_field(self, t, coord=None):
+    def build_field(self, t, coord=None, keep_edge=False):
         """
         Populate the field data using the analytical solution.
 
         :param t:
         :param coord: If None, take the coordinate from the numerical solver.
         :return: None
         """
+
         if None is coord:
-            coord = self.svr.coord[::2]  # Use the numerical solver.
+            _ = self.svr.ncoord
+            if keep_edge:
+                self.svr.xindices = np.linspace(
```

Comment:
The `xindices` attribute should be created in the initialization helper `Euler1DSolver.init_solver()` of the object `self.svr`: https://github.com/solvcon/modmesh/blob/214b4743bae1995cf7c451855880a397eae559fc/modmesh/onedim/euler1d.py#L36

Reply:
Do not add attributes outside the class.  It will be very difficult for a maintainer to keep track of where attributes are added, if they are not created in an organized way.

Reply:
> Do not add attributes outside the class. It will be very difficult for a maintainer to keep track of where attributes are added, if they are not created in an organized way.

@yungyuc I agree that we shouldn't add attributes outside the class, but I failed to create the `xindices` in the Euler1DCore https://github.com/solvcon/modmesh/blob/214b4743bae1995cf7c451855880a397eae559fc/modmesh/onedim/euler1d.py#L38
I tried to add the `xindices` in 
https://github.com/solvcon/modmesh/blob/214b4743bae1995cf7c451855880a397eae559fc/cpp/modmesh/onedim/Euler1DCore.cpp#L45 
like `m_coord` in
https://github.com/solvcon/modmesh/blob/214b4743bae1995cf7c451855880a397eae559fc/cpp/modmesh/onedim/Euler1DCore.cpp#L52
but I'm confused how to add it because I cannot decide its size in the begining. 

Or, maybe I misunderstand what you want me to do? Can you give me some suggestion?